internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -922,13 +950,34 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(log logrus.F
 		var replicas int32 = 1
 		deployment.Spec.Replicas = &replicas
 		deployment.Spec.Strategy = deploymentStrategy
+
+		// Handle our hashed configMap(s)
+		if deployment.Spec.Template.Annotations == nil {
+			deployment.Spec.Template.Annotations = make(map[string]string)
+		}
+		deployment.Spec.Template.Annotations[assistedConfigHashAnnotation] = assistedConfigHash
+		if mirrorConfigHash != "" {
+			deployment.Spec.Template.Annotations[mirrorConfigHashAnnotation] = mirrorConfigHash
+		}
+		if userConfigHash != "" {
+			runningUserConfigHash, ok := deployment.Spec.Template.Annotations[userConfigHashAnnotation]
+			if ok {
+				if runningUserConfigHash != userConfigHash {
+					msg := fmt.Sprintf("User provided and running configuration differ on assisted-service deployment. Remove the '%s' annotation from deployment to force restart", userConfigHashAnnotation)
+					r.Recorder.Event(instance, "Warning", "UserConfigMismatch", msg)
```

Comment:
@mhrivnak Same as above...This feels like an appropriate time to emit an event. What do you think?